lib/ui/ui_dart_state.cc

```diff
@@ -24,6 +24,67 @@ using tonic::ToDart;
 
 namespace flutter {
 
+namespace {
+void LogUnhandledException(Dart_Handle exception_handle,
+                           Dart_Handle stack_trace_handle) {
+  const std::string error =
+      tonic::StdStringFromDart(Dart_ToString(exception_handle));
+  const std::string stack_trace =
+      tonic::StdStringFromDart(Dart_ToString(stack_trace_handle));
+
+  auto state = UIDartState::Current();
+  FML_DCHECK(state);
+  if (state && UIDartState::Current()->unhandled_exception_callback()) {
+    auto callback = UIDartState::Current()->unhandled_exception_callback();
+    if (callback(error, stack_trace)) {
+      return;
+    }
+  }
+
+  // Either the exception handler was not set or it could not handle the
+  // error, just log the exception.
+  FML_LOG(ERROR) << "Unhandled Exception: " << error << std::endl
+                 << stack_trace;
+}
+
+void ReportUnhandledException(Dart_Handle exception_handle,
+                              Dart_Handle stack_trace_handle) {
+  // Hooks.dart will call the error handler on PlatformDispatcher if it is
+  // not null. If it is null, returns false, fall into the !handled branch
+  // below and log.
+  // If it is not null, defer to the return value of that closure
+  // to determine whether to report via logging.
+  bool handled = false;
+  auto state = UIDartState::Current();
+  FML_DCHECK(state);
```

Comment:
Remove this assert.

It's possible for a `CheckAndHandleError` to fail midway through isolate initialization before the isolate is bound to its `Dart_IsolateData`/`UIDartState::Current`.  If so, the error should be sent to `LogUnhandledException`.

The `FML_DCHECK(state)` in `LogUnhandledException` should also be removed.

Reply:
Done